docs(v1.7.0): add "Auto-Balance Pressured Disks" in important notes (#968)
longhorn/website: content/docs/1.7.0/deploy/important-notes/index.md

> **Note:**  In rare circumstances, it is possible for the failover to become deadlocked. This happens if the NFS server pod creation is blocked by a recovery action that is itself blocked by the failover-in-process state.  If the feature is enabled, and a failover takes more than a minute or two, it is probably stuck in this situation.  There is an explanation and a workaround in [RWX Volume Fast Failover](../../high-availability/rwx-volume-fast-failover).

### Support Configurable Timeout for Replica Rebuilding and Snapshot Cloning

> Since Longhorn v1.7.0, configurable timeouts for replica rebuilding and snapshot cloning are supported. Prior to v1.7.0, the timeout for replica rebuilding was capped at 24 hours, which could cause failures for large volumes in slow bandwidth environments. Now, the timeout is still 24 hours by default but can be adjusted to accommodate different environments. More information can be found [Settings Reference](../../references/settings/#long-grpc-timeout).

Reviewer: ```suggestion
Since Longhorn v1.7.0, configurable timeouts for replica rebuilding and snapshot cloning are supported. Prior to v1.7.0, the timeout for replica rebuilding was capped at 24 hours, which could cause failures for large volumes in slow bandwidth environments. Now, the timeout is still 24 hours by default but can be adjusted to accommodate different environments. More information can be found in the [Settings Reference](../../references/settings/#long-grpc-timeout).
```

To address the single point of failure (SPOF) issue with backing images, high availability for backing images was introduced in Longhorn v1.7.0. For more information, please see [Backing Image](../../advanced-resources/backing-image/backing-image/#number-of-copies).

## Scheduling

### Auto-Balance Pressured Disks

> In Longhorn v1.7.0, the replica auto-balancing feature was enhanced to address disk space pressure from growing volumes, introducing a new setting, `replica-auto-balance-disk-pressure-percentage`, that allows users to set a threshold for automatic action. This update reduces the need for manual intervention by automatically rebalancing replicas during disk pressure and improves performance by enabling faster replica rebuilds using local file copying. For more information, see the [setting](../../references/settings#replica-auto-balance-disk-pressure-threshold-) and this [link](https://github.com/longhorn/longhorn/issues/4105).

Reviewer: ```suggestion
In Longhorn v1.7.0, the replica auto-balancing feature was enhanced to address disk space pressure from growing volumes. This introduces a new setting, `replica-auto-balance-disk-pressure-percentage`, that allows users to set a threshold for automatic action. This update reduces the need for manual intervention by automatically rebalancing replicas during disk pressure. It also improves performance by enabling faster replica rebuilds using local file copying. For more information, see the [setting](../../references/settings#replica-auto-balance-disk-pressure-threshold-) and this [link](https://github.com/longhorn/longhorn/issues/4105).
```

## Networking

### Storage Network Support for Read-Write-Many (RWX) Volumes

Starting with Longhorn v1.7.0, the [storage network](../../advanced-resources/deploy/storage-network/) supports RWX volumes. However, the network's reliance on Multus results in a significant restriction.

Multus networks operate within the Kubernetes network namespace, so Longhorn can mount NFS endpoints only within the CSI plugin pod container network namespace. Consequently, NFS mount connections to the Share Manager pod become unresponsive when the CSI plugin pod restarts. This occurs because the namespace in which the connection was established is no longer available.

Longhorn circumvents this restriction by providing the following settings:

> - [Storage Network For RWX Volume Enabled](../../references/settings#storage-network-for-rwx-volume-enabled): When this setting is disabled, the storage network applies only to RWO volumes. The NFS client for RWX volumes is mounted over the cluster network in the host network namespace. This means that restarting the CSI plugin pod does not affect the NFS mount connections
> - [Automatically Delete Workload Pod when The Volume Is Detached Unexpectedly](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly): When the RWX volumes are created over the storage network, this setting actively deletes RWX volume workload pods when the CSI plugin pod restarts. This allows the pods to be remounted and prevents dangling mount entries.

Reviewer: ```suggestion
- [Storage network for RWX volume enabled](../../references/settings#storage-network-for-rwx-volume-enabled): When this setting is disabled, the storage network applies only to RWO volumes. The NFS client for RWX volumes is mounted over the cluster network in the host network namespace. This means that restarting the CSI plugin pod does not affect the NFS mount connections
- [Automatically delete workload pod when the volume is detached unexpectedly](../../references/settings#automatically-delete-workload-pod-when-the-volume-is-detached-unexpectedly): When the RWX volumes are created over the storage network, this setting actively deletes RWX volume workload pods when the CSI plugin pod restarts. This allows the pods to be remounted and prevents dangling mount entries.
```
